ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoverySuspendTest.java

```diff
@@ -0,0 +1,205 @@
+import com.arjuna.ats.jta.recovery.XAResourceOrphanFilter;
+import com.hp.mwtests.ts.jta.common.BytemanControlledXAResource;
+import com.hp.mwtests.ts.jta.common.BytemanControlledXAResourceRecovery;
+import jakarta.transaction.HeuristicMixedException;
+import jakarta.transaction.HeuristicRollbackException;
+import jakarta.transaction.RollbackException;
+import jakarta.transaction.SystemException;
+import org.jboss.byteman.contrib.bmunit.BMScript;
+import org.jboss.byteman.contrib.bmunit.BMScripts;
+import org.jboss.byteman.contrib.bmunit.BMUnitConfig;
+import org.jboss.byteman.contrib.bmunit.WithByteman;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javax.transaction.xa.XAException;
+import javax.transaction.xa.XAResource;
+import javax.transaction.xa.Xid;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+@WithByteman
+@BMUnitConfig(debug = true)
+public class RecoverySuspendTest {
+
+    private static RecoveryManager _manager;
+
+    private final static String[] modules = {
+            // the Recovery Module to test
+            "com.arjuna.ats.internal.jta.recovery.arjunacore.XARecoveryModule"
+    };
+
+    private static RecoveryEnvironmentBean _recoveryConfig;
+    private static JTAEnvironmentBean _jtaEnvironmentBean;
+
+    // fields read with byteman
+    private final static int periodicRecoveryPeriod = 5;
+    private final static int recoveryBackoffPeriod = 1;
+
+    @BeforeAll
+    public static void beforeClass() {
+
+        _recoveryConfig = recoveryPropertyManager.getRecoveryEnvironmentBean();
+        _jtaEnvironmentBean = jtaPropertyManager.getJTAEnvironmentBean();
+
+        // Let's go quick on this
+        _recoveryConfig.setRecoveryBackoffPeriod(recoveryBackoffPeriod);
+        _recoveryConfig.setPeriodicRecoveryPeriod(periodicRecoveryPeriod);
+        // Only one millisecond to stale record as soon as possible
+        _jtaEnvironmentBean.setOrphanSafetyInterval(1);
+        // don't sign off until the store is empty
+        _recoveryConfig.setWaitForWorkLeftToDo(true);
+
+        // the orphan filter
+        XAResourceOrphanFilter xaResourceOrphanFilter = new DummyXAResourceOrphanFilter(XAResourceOrphanFilter.Vote.ROLLBACK);
+        _jtaEnvironmentBean.setXaResourceOrphanFilters(List.of(xaResourceOrphanFilter));
+        // the test set of modules
+        _recoveryConfig.setRecoveryModuleClassNames(Arrays.asList(modules));
+
+        _jtaEnvironmentBean.setXaResourceRecoveryClassNames(
+                List.of(BytemanControlledXAResourceRecovery.class.getName()));
+
+        // obtain a new RecoveryManager with the above config:
+        _manager = RecoveryManager.manager(RecoveryManager.INDIRECT_MANAGEMENT);
+        // recovery can start
+        _manager.startRecoveryManagerThread();
```

Comment:
Do I recall that recently you were proposing to remove this call as it was not needed?

Reply:
If it is needed, that might help with the asserting rollback was not called thing I was talking about below (i.e. that rollback was not called during the `createXATransaction` call...

Reply:
> Do I recall that recently you were proposing to remove this call as it was not needed?

Yep, you do recall correctly. I'll remove that instruction. Thanks for spotting it